src/dashboard/src/main/management/commands/rebuild_aip_index_from_storage_service.py

```diff
@@ -0,0 +1,326 @@
+        filtered_aic_count = len(filtered_aics)
+
+        aic_indexed_count = 0
+        for aic in filtered_aics:
+            index_success = self.process_package(
+                es_client, aic, temp_dir, delete_before_reindexing, is_aic=True
+            )
+            if index_success is True:
+                aic_indexed_count += 1
+            else:
+                packages_not_indexed.append(aic["uuid"])
+
+        self.info("Cleaning up")
+
+        shutil.rmtree(temp_dir)
+
+        if packages_not_indexed:
+            indexing_stats = "{0} of {1} AIPs and {2} of {3} AICs".format(
+                aip_indexed_count,
+                filtered_aip_count,
+                aic_indexed_count,
+                filtered_aic_count,
+            )
+            self.info(
+                "Indexing complete. Indexed {0}. Packages not indexed: {1}.".format(
+                    indexing_stats, ", ".join(packages_not_indexed)
+                )
+            )
+        else:
+            self.info(
+                "Indexing complete. Indexed {0} AIPs and {1} AICs".format(
+                    aip_indexed_count, aic_indexed_count
+                )
+            )
+
+    def process_package(
+        self, es_client, package_info, temp_dir, delete_before_reindexing, is_aic=False
+    ):
+        """Index package in 'aips' and 'aipfiles' indices.
+
+        :param es_client: Elasticsearch client.
+        :param package_info: Package info dict returned by Storage Service.
+        :param temp_dir: Path to temporary directory for downloaded METS files.
+        :param delete_before_reindexing: Boolean of whether to delete package from indices prior to reindexing.
+        :is_aic: Optional boolean to indicate if package being indexed is an AIC.
+
+        :returns: Boolean indicating success.
+        """
+        uuid = package_info["uuid"]
+
+        # Download the AIP METS file to a temporary directory.
+        mets_relative_path = am.relative_path_to_aip_mets_file(
+            package_info["uuid"], package_info["current_path"]
+        )
+        mets_filename = os.path.basename(mets_relative_path)
+        mets_download_path = os.path.join(temp_dir, mets_filename)
+        storageService.extract_file(uuid, mets_relative_path, mets_download_path)
+
+        if not os.path.isfile(mets_download_path):
+            error_message = "Unable to download AIP METS file from Storage Service"
+            self.error(
+                "Error indexing package {0}. Details: {1}".format(uuid, error_message)
+            )
+            return False
+
+        aips_in_aic = None
+        if is_aic is True:
+            aips_in_aic = get_aips_in_aic(package_info, mets_download_path, temp_dir)
+
+        package_name = am.package_name_from_path(
+            package_info["current_path"], remove_uuid_suffix=True
+        )
+
+        if delete_before_reindexing is True:
+            es.delete_aip(es_client, uuid)
```

Comment:
Is it worth logging which package is about to be deleted like `rebuild_elasticsearch_aip_index_from_files.py` does?